### PR TITLE
Enable reindex via ui

### PR DESCRIPTION
**Pull request by LATaylor-guardian**
This PR adds the following:
 - Ability to initiate a reindex for a content source via the UI.
 - When a reindex is initiated, a progress indicator is displayed on the UI. This does not currently update itself with progress (this is to come).
 - Ability to cancel a reindex. When cancelling a reindex, the progress indicator will disappear and the UI will update the job history with the newly cancelled item.

### CODE REVIEW

File: app/com/gu/floodgate/contentsource/ContentSourceApi.scala [hunk 1]
```diff
@@ -66,12 +70,27 @@ class ContentSourceApi(contentSourceService: ContentSourceService, reindexServic
     }
   }
 
+  def cancelReindex(id: String) = Action.async { implicit request =>
+    reindexService.cancelReindex(id) map { happyOrError =>
+      happyOrError match {
+        case Good(_) => Ok
+        case Bad(error) => BadRequest(Json.toJson(ErrorResponse(error.message)))
```
Comment (cb372) [inline app/com/gu/floodgate/contentsource/ContentSourceApi.scala]: Splitting hairs, but I'm not sure if `BadRequest` is the best response code here. It's hard to say whether the failure is the client's or the server's fault, so maybe `InternalServerError` would be more appropriate.

**Comment by cb372** [comment]
:+1: Looks good. You're a React master these days!

File: app/com/gu/floodgate/contentsource/ContentSourceApi.scala (hunk 1 above)
Comment (LATaylor-guardian) [inline app/com/gu/floodgate/contentsource/ContentSourceApi.scala]: Hmmm. There are three error outcomes here:

1. The request to cancel the reindex from the content source tool fails and therefore this is an internal server error.
2. No content source exists for the given content source and therefore this is a bad request.
3. There is no running reindex for this content source and therefore this is a bad request.

I think CustomError type impedes here. Maybe I can group those specific errors under a new type, CancelReindexErrors and match on that to resolve the error response more specifically.

Comment (LATaylor-guardian) [inline app/com/gu/floodgate/contentsource/ContentSourceApi.scala]: In fact, I'm going to do this in a separate PR as there are probably more cases than just this.
